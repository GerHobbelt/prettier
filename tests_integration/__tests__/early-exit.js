"use strict";

const prettier = require("../..");
const runPrettier = require("../runPrettier");

test("show version with --version", () => {
  const result = runPrettier("cli/with-shebang", ["--version"]);

  expect(result.stdout).toBe(prettier.version + "\n");
  expect(result.status).toEqual(0);
});

test("show usage with --help", () => {
  const result = runPrettier("cli", ["--help"]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(0);
});

test("throw error with --write + --debug-check", () => {
  const result = runPrettier("cli", ["--write", "--debug-check"]);

  expect(result.stderr).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("throw error with --find-config-path + multiple files", () => {
  const result = runPrettier("cli", ["--find-config-path", "abc.js", "def.js"]);

  expect(result.stderr).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("throw error and show usage with something unexpected", () => {
  const result = runPrettier("cli", [], { isTTY: true });

  expect(result.stdout).toMatchSnapshot();
  expect(result.stderr).toMatchSnapshot();
  expect(result.status).not.toEqual(0);
});
