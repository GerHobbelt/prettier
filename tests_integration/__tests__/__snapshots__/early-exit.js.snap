// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show usage with --help 1`] = `
"Usage: prettier [opts] [filename ...]

Available options:
  --write                  Edit the file in-place. (Beware!)
  --list-different or -l   Print filenames of files that are different from Prettier formatting.
  --config                 Path to a prettier configuration file (.prettierrc, package.json, prettier.config.js).
  --config-precedence <cli-override|file-override|prefer-file>
                           Defines how config file should be evaluated in combination of CLI options
                           cli-override  | default config => config file => CLI options
                           file-override | default config => CLI options => config file
                           prefer-file   | default config => config file (if config file is found) or
                                           default config => CLI options (if no config file is found)
                           Defaults to cli-override
  --no-config              Do not look for a configuration file.
  --find-config-path <path>
                           Finds and prints the path to a configuration file for a given input file.
  --ignore-path <path>     Path to a file containing patterns that describe files to ignore.
                           Defaults to ./.prettierignore.
  --require-pragma
                           Require either '@prettier' or '@format' to be present in the file's first docblock comment in order for it to be formatted.
  --stdin                  Read input from stdin.
  --stdin-filepath         Path to the file used to read from stdin.
  --print-width <int>      Specify the length of line that the printer will wrap on. Defaults to 80.
  --tab-width <int>        Specify the number of spaces per indentation-level. Defaults to 2.
  --use-tabs               Indent lines with tabs instead of spaces.
  --no-semi                Do not print semicolons, except at the beginning of lines which may need them.
  --single-quote           Use single quotes instead of double quotes.
  --no-bracket-spacing     Do not print spaces between brackets.
  --jsx-bracket-same-line  Put > on the last line instead of at a new line.
  --trailing-comma <none|es5|all>
                           Print trailing commas wherever possible when multi-line. Defaults to none.
  --parser <flow|babylon|typescript|postcss|json|graphql>
                           Specify which parse to use. Defaults to babylon.
  --cursor-offset <int>    Print (to stderr) where a cursor at the given position would move to after formatting.
                           This option cannot be used with --range-start and --range-end
  --range-start <int>      Format code starting at a given character offset.
                           The range will extend backwards to the start of the first line containing the selected statement.
                           This option cannot be used with --cursor-offset.
                           Defaults to 0.
  --range-end <int>        Format code ending at a given character offset (exclusive).
                           The range will extend forwards to the end of the selected statement.
                           This option cannot be used with --cursor-offset.
                           Defaults to Infinity.
  --no-color               Do not colorize error messages.
  --with-node-modules      Process files inside 'node_modules' directory.
  --version or -v          Print Prettier version.


"
`;

exports[`throw error with --find-config-path + multiple files 1`] = `
"Cannot use --find-config-path with multiple files
"
`;

exports[`throw error with --write + --debug-check 1`] = `
"Cannot use --write and --debug-check together.
"
`;
